src/Microsoft.ML/PredictionModel.cs

```diff
@@ -29,6 +30,39 @@ internal Runtime.EntryPoints.TransformModel PredictorModel
             get { return _predictorModel; }
         }
 
+        /// <summary>
+        /// Returns labels that correspond to indices of the score array in the case of 
+        /// multi-class classification problem.
+        /// </summary>
+        /// <param name="mapping">Label to score mapping</param>
+        /// <param name="scoreColumnName">Name of the score column</param>
+        /// <returns></returns>
+        public bool TryGetScoreLabelMapping(out string[] mapping, string scoreColumnName = DefaultColumnNames.Score)
+        {
+            mapping = null;
+            ISchema schema = _predictorModel.OutputSchema;
+            int colIndex = -1;
+            if (!schema.TryGetColumnIndex(scoreColumnName, out colIndex))
+                return false;
+            
+            int expectedLabelCount = schema.GetColumnType(colIndex).AsVector.ValueCount;
```

Comment:
>AsVector [](start = 68, length = 8)

`AsVector` can be null, in the case where the type is not vector. REplace `AsVector.ValueCount` with `VectorSize`. Below this you will need to return `false` if `VectorSize` is not a positive number.